basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -1151,53 +1151,94 @@ namespace basic_autonomy
 
-            //Compute points to local downtracks
-            std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(future_geom_points);
-
-            auto total_step_along_curve = static_cast<int>(downtracks.back() /detailed_config.curve_resample_step_size);
-
-            double scaled_steps_along_curve = 0.0; //from 0 (start) to 1 (end) for the whole trajectory
-
-            for(int steps_along_curve = 0; steps_along_curve < total_step_along_curve; steps_along_curve++){
-                lanelet::BasicPoint2d p = (*fit_curve)(scaled_steps_along_curve);
+            // Compute downtracks for original points (for interpolation reference)
+            std::vector<double> original_downtracks = carma_wm::geometry::compute_arc_lengths(future_geom_points);
+
+            // Now create resampled points using the spline
+            auto total_step_along_curve = static_cast<int>(original_downtracks.back() / detailed_config.curve_resample_step_size);
+            if (total_step_along_curve == 0) {
+                RCLCPP_WARN_STREAM(rclcpp::get_logger(BASIC_AUTONOMY_LOGGER),
+                "Available distance to resample is less than curve_resample_step_size. "
+                "Only considering the last point of the target destination to generate trajectory."
+                );
+                total_step_along_curve = 1; // This also avoids division by zero
+            }
 
-                all_sampling_points.push_back(p);
+            std::vector<lanelet::BasicPoint2d> resampled_points;
+            resampled_points.reserve(total_step_along_curve);
 
+            double scaled_steps_along_curve = 0.0; // from 0 (start) to 1 (end) for the whole trajectory
+            for(int step = 0; step < total_step_along_curve; step++){
                 scaled_steps_along_curve += 1.0 / total_step_along_curve;
+                lanelet::BasicPoint2d p = (*fit_curve)(scaled_steps_along_curve);
+                resampled_points.push_back(p);
             }
-            RCLCPP_DEBUG_STREAM(rclcpp::get_logger(BASIC_AUTONOMY_LOGGER), "Got sampled points with size:" << all_sampling_points.size());
-
-            std::vector<double> final_yaw_values = carma_wm::geometry::compute_tangent_orientations(future_geom_points);
-            if(final_yaw_values.size() > 0) {
-                final_yaw_values[0] = state.orientation; // Set the initial yaw value based on the initial state
+            RCLCPP_DEBUG_STREAM(rclcpp::get_logger(BASIC_AUTONOMY_LOGGER), "Got resampled points with size:" << resampled_points.size());
+
+            // Compute downtracks for the resampled points
+            std::vector<double> resampled_downtracks = carma_wm::geometry::compute_arc_lengths(resampled_points);
+
+            // Interpolate speeds based on downtracks
+            std::vector<double> resampled_speeds;
+            resampled_speeds.reserve(resampled_points.size());
+
+            for (const auto& downtrack : resampled_downtracks) {
```

Comment:
Do we need this manual speed interpolation? Seems like moving average filter should smooth this out?

Reply:
Sorry I forgot we discussed this offline. I remove the interpolation.